lib/extractor/oci-archive/layer.ts

```diff
@@ -220,3 +231,40 @@ function isImageIndexFile(json: any): boolean {
 function isMainIndexFile(name: string): boolean {
   return name === "index.json";
 }
+
+function getBestMatchForPlatform<
+  T extends {
+    platform?: OciPlatformInfo;
+    os?: string;
+    architecture?: string;
+    variant?: string;
+  },
+>(manifests: T[], platform: string): T | undefined {
```

Comment:
I'm not super happy with this type. The function handles both image manifests and configs and uses the exact same logic to compare platform info, however the shape of manifests and configs is different so I used a generic and extended it to include the different options.